### PR TITLE
JOSS paper

### PR DESCRIPTION
**Pull request by mdmurbach**
I think we're getting close on the JOSS paper! A few last remaining loose ends I can think of before we merge (feel free to add other todos below):

- [x] @nealde can you add your contact info (unless you don't want to add your email/orcid)
- [x] should we add any references to EIS being used in the last section of the background section? https://github.com/ECSHackWeek/impedance.py/blob/paper/JOSS/joss-paper/paper.md#background

### CODE REVIEW

**Comment by BGerwe** [comment]
Yeah, that's a good idea. I'm sure we can find a review paper that covers all of those bases.

**Comment by mdmurbach** [comment]
compiled version: http://res.cloudinary.com/hju22ue2k/image/upload/v1584240701/lzoowji4tcppdknwkxj1.pdf

- [ ] change 'data preprocessing' to 'preprocessing' so this first one fits on a single line 😂 

image

**Comment by mdmurbach** [comment]
Also, we should change the example circuit to be more physically meaningful (re: Dan’s point here, https://github.com/ECSHackWeek/impedance.py/issues/80#issuecomment-593057731)

- [ ] update example circuit string

File: joss-paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+# Statement of Need
+To date, typical impedance analysis solutions have relied on either instrument-specific, proprietary software or ad hoc, lab-specific code written for internal use. In addition to making reproducible analysis difficult, these solutions can be restrictive to defining custom circuit elements and processing large datasets. By providing an open-source, community-driven package for the full impedance analysis pipeline from data management to parameter extraction and publication ready figures, `impedance.py` seeks to encourage reproducible, easy-to-use, and transparent analysis. In addition to decades of electrochemical research, many new methods for validating [@schonleber_method_2014] and analyzing [@murbach_analysis_2018; @buteau_analysis_2019] impedance spectra have been developed by researchers. By lowering the barrier to use tried-and-true methods along side cutting-edge analytical techniques via a consistent interface, `impedance.py` also serves to grow as a community repository of best-practices while facilitating the adoption of new techniques.
+
+# Current `impedance.py` functionality
+
+## Data Ingestion
```
Comment (nealde) [inline joss-paper/paper.md]: minor formatting: we should decide if we want to capitalize both words or only the first one

**Comment by nealde** [comment]
I tried looking for where your contact information was, but couldn't find it.  here's mine:
email: ndawsonelli@gmail.com
orcid: https://orcid.org/0000-0003-4559-7309

File: joss-paper/paper.md (hunk 1 above)
Comment (BGerwe) [reply joss-paper/paper.md]: I vote we go with only the first word capitalized. I'll also change this to `## Data preprocessing` to be consistent with the workflow outlined above.

**Comment by mdmurbach** [comment]
@nealde it’s at the very top in the metadata (you might have to look at the raw version of the .md file to see it on GitHub: https://raw.githubusercontent.com/ECSHackWeek/impedance.py/paper/JOSS/joss-paper/paper.md). Is “PayScale, Inc.” (capitalization?) the right affiliation?

**Comment by BGerwe** [comment]
@mdmurbach I'm working on adding his info right now and updating the example to follow [Dan's suggestion](https://github.com/ECSHackWeek/impedance.py/issues/80#issuecomment-593057731)

**Comment by mdmurbach** [comment]
Awesome! Thanks, @BGerwe!

File: joss-paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+circuit = CustomCircuit('L_0-R_0-p(R_1,CPE_1)-p(R_2,CPE_2)-Wo_1',
+                        initial_guess=initial_guess)
+
+# 3. fitting the circuit to the data
+circuit.fit(f, Z)
+print(circuit)
+```
+
+```text
+Circuit string: L0-R0-p(R1,E1)-p(R2,E2)-W1
```
Comment (BGerwe) [inline joss-paper/paper.md]: This circuit string is showing the old element naming format. I'm guessing this will be fixed if we re-run the example?

**Comment by lktsui** [comment]
Hi Matt,

I made a few edits a while back but I don't think they got included in this one. 

https://github.com/ECSHackWeek/impedance.py/pull/92

File: joss-paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+# Background
+
+Electrochemical impedance spectroscopy (EIS) is a powerful technique for noninvasively probing the physicochemical processes governing complex electrochemical systems by measuring the relationship between voltage and current (i.e. the impedance) as a function of frequency [@orazem_electrochemical_2008]. Analysis of EIS spectra often involves the construction and fitting of models to represent the system response. Equivalent circuit models, a popular method for interpreting impedance spectra, represent processes in an electrochemical system with elements in a circuit allowing quantitative characterization of the data. EIS inherently requires the system response is linear, causal and stable over the measurement time. As such, researchers must also ensure their data complies with these assumptions – often through some implementation of Kramers-Kronig analysis. Following the approach outlined above, workers have applied EIS to problems ranging from  measuring the dielectric properties of biological systems to corrosion rates of coated metals. More recently, EIS has been widely adopted for studying processes governing batteries and fuel cells.
```
Comment (lktsui) [inline joss-paper/paper.md]: We should add references here for the stated applications.

Can't find the specific paper for dielectric properties of biological systems. Anyone know which paper this is talking about?

Corrosion: https://www.sciencedirect.com/science/article/pii/0013468695004408 

Batteries: https://www.sciencedirect.com/science/article/abs/pii/S0378775399003511 + https://www.sciencedirect.com/science/article/pii/S0013468605007899

Fuel Cells: https://iopscience.iop.org/article/10.1149/1.1836485/meta + https://www.sciencedirect.com/science/article/pii/S0013468698001388

Comment (BGerwe) [reply joss-paper/paper.md]: Thanks for those references! I'll add them to my local version now. The measurement of dielectric properties of biological systems was mentioned in the intro of Mark Orazem's EIS book, so I'll try to find the reference used in that book.

Comment (BGerwe) [reply joss-paper/paper.md]: These references were added in [this commit](https://github.com/ECSHackWeek/impedance.py/pull/91/commits/78fd7c5c291dba03491e0dc59b898838539d4c28).

**Comment by BGerwe** [comment]
@mdmurbach I changed the example circuit according to Dan's suggestion (I think?), but can you double check for me? I had trouble re-generating a high resolution .png because I don't have the right packages installed for altair. I can redo that in a later commit, or I'll leave it up to you.

**Comment by mdmurbach** [comment]
Sounds good! Thanks 😊 Good to merge after I add the png?

**Comment by BGerwe** [comment]
> compiled version: http://res.cloudinary.com/hju22ue2k/image/upload/v1584240701/lzoowji4tcppdknwkxj1.pdf
> 
> * [ ]  change 'data preprocessing' to 'preprocessing' so this first one fits on a single line 😂
> 
> image

Just realized I never changed this part. I'll fix it quickly and it should be good after that.

**Comment by BGerwe** [comment]
@mdmurbach It should be good to go now! :rocket:

**Pull request merged**